docs/4.0/ssh_gsuite.md

```diff
@@ -0,0 +1,158 @@
+# SSH Authentication with G Suite
+
+This guide will cover how to configure [G Suite](https://gsuite.google.com) to issue
+SSH credentials to specific groups of users. When used in combination with role
+based access control (RBAC) it allows SSH administrators to define policies
+like:
+
+* Only members of "DBA" Google group can SSH into machines running PostgreSQL.
+* Developers must never SSH into production servers.
+* ... and many others.
+
+!!! warning "Version Warning":
+    This guide requires a commercial edition of Teleport. The open source
+    edition of Teleport only supports [Github](admin-guide/#github-oauth-20) as
+    an SSO provider.
+
+
+## Configure G Suites
+
+1. Obtain OAuth 2.0 credentials  [https://developers.google.com/identity/protocols/OpenIDConnect](https://developers.google.com/identity/protocols/OpenIDConnect)
+
+2. Create a new Project.
+![Create New Project](img/gsuite/gsuite-1-new-project.png)
+
+3. Select OAuth client ID.
+![Create OAuth Creds](img/gsuite/gsuite-2-created-creds.png)
+
+4. Make Application Type Public & Setup Domain Verification 
+![Setup Application Type](img/gsuite/gsuite-3-oauth.png)
+
+5. Copy Oauth Client ID and Client Secret for YAML Below.
+![Copy Client Secret](img/gsuite/gsuite-5-copy-client-id.png)
+
+6. To allow for fine grained role based access control (RBAC) you'll first need to
+open an admin API so Teleport can assign groups to roles. Allow Admin SDK via [google.com/apis/api/admin.googleapis.com/overview](https://console.developers.google.com/apis/api/admin.googleapis.com/overview)
+![Turn on Admin SDK](img/gsuite/gsuite-4-admin-sdk.png)
+
+
+
+## Create a OIDC Connector
+
+Now, create a OIDC connector [resource](admin-guide#resources).
+Write down this template as `gsuite-connector.yaml`:
+
+```yaml
+kind: oidc
+version: v2
+metadata:
+  name: gsuite
+spec:
+  redirect_url: https://localhost:3080/v1/webapi/oidc/callback
+  client_id: exampleclientid11234.apps.googleusercontent.com
+  client_secret: examplesecret
+  issuer_url: https://accounts.google.com
+  scope: ['https://www.googleapis.com/auth/admin.directory.group.readonly', 'openid', 'email']  
+  claims_to_roles:
+    - {claim: "groups", value: "admin@example.com", roles: ["admin"]}
+```
+
+!!! important 
+    The groups will be fetched only if admins include special auth scope https://www.googleapis.com/auth/admin.directory.group.readonly in the scopes of the connector as shown in the example above.
+
+
+Create the connector using `tctl` tool:
+
+```bsh
+$ tctl create gsuite-connector.yaml
+```
+
+## Create Teleport Roles
+
+We are going to create 2 roles, privileged role admin who is able to login as
```

Comment:
It might be better to make this a bullet-pointed list for clarity:

```
We are going to create 2 roles:
- privileged role admin who is able to login as root and is capable of administrating the cluster
- non-privileged dev
```